### DISCUSSION TITLE
Converting an iterator over statements to ntriples

### DISCUSSION THREAD

**Discussion by D063520** [eclipse-rdf4j/rdf4j#4365] [Q&A] [answered]
Hi,

I have an iterator over statements:

```
Statement st = statements.next();
String ntriple = ......
```
And I would like to get back a string that contains the Ntriple representation of the statement st.

Can someone help me out?

**Comment by frensjan** [comment, marked as answer]
```java
Iterator<Statement> statements = ...;

StringWriter sw = new StringWriter();
RDFWriter writer = Rio.createWriter(RDFFormat.NTRIPLES, sw);
writer.startRDF();
statements.forEachRemaining(writer::handleStatement);
writer.endRDF();
String ntriples = sw.toString();
```

This would give you a string with _all_ the statements from the iterator. If you _must_ have a string per statement, you would probably need to create a writer per statement then.

```java
Iterator<Statement> statements = ...;

while (statements.hasNext()) {
    StringWriter sw = new StringWriter();
    RDFWriter writer = Rio.createWriter(RDFFormat.NTRIPLES, sw);
    writer.startRDF();
    writer.handleStatement(statements.next());
    writer.endRDF();
    String ntriple = sw.toString();
}
```

**Reply by D063520** [reply]
thank you!

**Comment by barthanssens** [comment]
Excellent answer from @frensjan 

FWIW, there is also a `NTriplesUtil` class that might be useful (https://rdf4j.org/javadoc/4.2.2/org/eclipse/rdf4j/rio/helpers/NTriplesUtil.html) in this specific case.
And if the `Statements` originate from a `Model` / `Iterable<Statement>`, there is possible shortcut in using `Rio.write(model, out, RDFFormat.NTRIPLES)`

**Reply by D063520** [reply]
thank you!
